File: src/main/java/com/heybys/oddments/domain/user/UserId.java
package com.heybys.oddments.domain.user;

import com.heybys.oddments.base.domain.LongTypeIdentifier;
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType;

public class UserId extends LongTypeIdentifier {

    public UserId(Long id) {
        super(id);
    }

    public static class UserIdJavaType extends LongTypeIdentifierJavaType<UserId> {
        public UserIdJavaType() {
            super(UserId.class);
        }
    }
}

File: src/main/kotlin/com/heybys/oddments/fooddelivery/domain/user/UserId.kt
package com.heybys.oddments.fooddelivery.domain.user

import com.heybys.oddments.base.domain.LongTypeIdentifier
import com.heybys.oddments.base.jpa.hibernate.LongTypeIdentifierJavaType

class UserId(id: Long) : LongTypeIdentifier(id) {

    class UserIdJavaType : LongTypeIdentifierJavaType<UserId>(UserId::class.java)
}
